### PR TITLE
Refactor namespace at rule

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Implement new `Url` type (spec https://www.w3.org/TR/css-values-4/#urls): 
1. Fix bugs with parsing around `url()`/`src()`
2. Supports modifiers
3. Keep original name `Url` token
4. Refactor `at-rules`, now they accept `Url` type according spec
5. Fix parsing `@namespace`
6. Fix some clippy warnings
7. Added `TODO` for future fixes

**BREAKING CHANGE:**

Yes, new type

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_parser/src/lexer/mod.rs [hunk 1]
```diff
@@ -542,7 +542,29 @@ where
                 }
                 // Otherwise, consume a url token, and return it.
                 _ => {
-                    return self.read_url();
+                    return self.read_url().map(|mut token| {
```
Comment (kdy1) [inline crates/swc_css_parser/src/lexer/mod.rs:545]: Why does `read_url` return an incomplete token?
I think accepting `name` as a parameter and returning something in a valid state is better

Comment (kdy1) [reply crates/swc_css_parser/src/lexer/mod.rs:545]: No partial-init is one of philosophies of rust.
Is there a drawback of this approach?

**Review by kdy1 — COMMENTED** [review]
Thank you! I have some questions about the changes.

File: crates/swc_css_parser/src/lexer/mod.rs (hunk 1 above)
Comment (alexander-akait) [reply crates/swc_css_parser/src/lexer/mod.rs:545]: Yep, initially I do it, but name in one function, but token creation in other, so I tried to keep them independent, but it is not problem and I can refactor it

Comment (kdy1) [reply crates/swc_css_parser/src/lexer/mod.rs:545]: Can you do the refactor? I think it's a better way to go.

Comment (alexander-akait) [reply crates/swc_css_parser/src/lexer/mod.rs:545]: Yes, I will do it in near future :+1:

Comment (kdy1) [reply crates/swc_css_parser/src/lexer/mod.rs:545]: Thank you! I think we should follow the philosophy of the language if we can.

Comment (alexander-akait) [reply crates/swc_css_parser/src/lexer/mod.rs]: @kdy1 Done :+1: Found couple TODO with `url`, I will fix them in the next PR

**Review by kdy1 — APPROVED** [review]
Thank you!

**Pull request merged**